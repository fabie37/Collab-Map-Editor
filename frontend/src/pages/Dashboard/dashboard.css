div.left-panel {
    width: 15%;
    height: 100vw;
    opacity: 0.8;
    background: #ffffff;
    margin-top: 0px;
    border-radius: 4px;
    padding: 30px;
    display: flex;
    flex-direction: column;
    margin-bottom: 50px;
    float: left;
}

Button.left-panel-menu-item {
    font-size: 20px;
    border: #ffffff;
    background-color: #ffffff;
    text-align: left;
    margin-top: 20px;
}

div.mapPanel {
    width: 50%;
    opacity: 0.8;
    background: #ffffff;
    margin-top: 50px;
    margin-bottom: 50px;
    border-radius: 4px;
    padding: 30px;
    display: flex;
    flex-direction: column;
    margin-left: 450px;
}

#testbutton{
    font-size: 20px;
    color: black;
    border: #ffffff;
    background-color: #ffffff;
    text-align: left;
    margin-top: 20px;
    text-decoration: none;
}

#testbutton:hover{
    background-color: rgb(43, 116, 226);
}

li.map{
    width: 300px;
    opacity: 100%;
    background: #e4e4e4;
    margin-top: 50px;
    margin-bottom: 50px;
    margin-left: 50px;
    border-radius: 4px;
    /*border: solid #000;*/
    /*border-width: 1 1px;*/
    box-shadow: 5px 5px 5px grey;
    padding: 30px;
    display: inline-flex;
    flex-direction: column;
}

li.map strong{
    opacity: 100%;
    font-size: 35px;
    color: black;
}

li.map span{
    opacity: 100%;
    font-size: 20px;
    color: black;
}

ul.events-list {
    width: 100%;
    list-style: none;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 20px;
    margin-bottom: 30px;
}

ul.events-list li {
    display: flex;
    flex-direction: column;
}

ul.events-list li header {
    width: 100%;
    height: 240px;
    background-size: contain;
    background-repeat: no-repeat;
    border-radius: 4px;
}

ul.events-list li strong{
    margin-top: 12px;
    font-size: 24px;
    color: #444;
}


div.filter-panel{
    display: flex;
    justify-content: space-between;
    margin-bottom: 50px;
}

ul.events-list header div {
    display: flex;
    flex-direction: row-reverse;
}

ul.notifications {
    list-style: none;
    margin-bottom: 16px;
}

ul.notifications li {
    font-size: 16px;
    line-height: 24px;

}

@media (max-width:768px){
    ul.events-list{
        grid-template-columns: 1fr;
    }
}