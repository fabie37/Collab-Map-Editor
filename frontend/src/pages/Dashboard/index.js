import React, { useEffect, useState, useMemo } from 'react';
import api from '../../services/api';
import moment from 'moment';
import { Button, ButtonGroup, Container, Alert, Dropdown, Nav, NavLink, 
    DropdownItem, DropdownMenu, DropdownToggle, ListGroup, ListGroupItem, Card, CardBody, CardImg, CardTitle, CardSubtitle, CardText } from 'reactstrap';
import socketio from 'socket.io-client';
import './dashboard.css'
import battle from '../../assets/battle.jpg'


export default function Dashboard({ history }) {
    const [events, setEvents] = useState([])
    const user = localStorage.getItem('user')
    const user_id = localStorage.getItem('user_id')

    const [rSelected, setRSelected] = useState(null)
    const [error, setError] = useState(false);
    const [success, setSuccess] = useState(false)
    const [messageHandler, setMessageHandler] = useState('')
    const [browseMapsPanel, setBrowseMapsPanel] = useState(false)
    const [myMapsPanel, setMyMapsPanel] = useState(false)

    useEffect(() => {
        getEvents()
    }, [])

    const socket = useMemo(
        () =>
            socketio('http://localhost:8000/', { query: { user: user_id } }),
        [user_id]
    );

    const filterHandler = (query) => {
        setRSelected(query)
        getEvents(query)
    }

    const myEventsHandler = async () => {
        try {
            setRSelected('myevents')
            const response = await api.get('/user/events', { headers: { user } })
            setEvents(response.data.events)
        } catch (error) {
            history.push('/login');
        }

    }

    const getEvents = async (filter) => {
        try {
            const url = filter ? `/dashboard/${filter}` : '/dashboard';
            const response = await api.get(url, { headers: { user } })

            setEvents(response.data.events)
        } catch (error) {
            history.push('/login');
        }

    };

    const deleteEventHandler = async (eventId) => {
        try {
            await api.delete(`/event/${eventId}`, { headers: { user: user } });
            setSuccess(true)
            setMessageHandler('The event was deleted successfully!')
            setTimeout(() => {
                setSuccess(false)
                filterHandler(null)
                setMessageHandler('')
            }, 2500)

        } catch (error) {
            setError(true)
            setMessageHandler('Error when deleting event!')
            setTimeout(() => {
                setError(false)
                setMessageHandler('')
            }, 2000)
        }
    }

    console.log("middle panel: ", browseMapsPanel)

    const browseMapsFunc = async() => {
        setBrowseMapsPanel(!browseMapsPanel)
        if (myMapsPanel){
            setMyMapsPanel(false)
        }
    }

    const myMapsFunc = async() => {
        setMyMapsPanel(!myMapsPanel)
        if (browseMapsPanel){
            setBrowseMapsPanel(false)
        }
    }

    return (
        <>
            <div className="left-panel">
                <ListGroup>
                    <Button id="testbutton" color="link" onClick={() => browseMapsFunc()}> Browse Maps </Button>
                    <Button id="testbutton" color="link" onClick={() => myMapsFunc()}> My Maps </Button>
                    <Button id="testbutton" color="link"> Shared with Me </Button>
                    <Button id="testbutton" color="link"> Help </Button>
                </ListGroup>
            </div>
            {
                browseMapsPanel ? (
                    <div className="mapPanel">
                        <h2>Browse Maps</h2>
                        <ul>
                            <li className="map">
                                <strong> Map </strong>
                                <img src={battle} width="200"/>
                                <span> Date </span>
                                <span> Description </span>
                            </li>

                            <li className="map">
                                <strong> Map </strong>
                                <img src={battle} width="200"/>
                                <span> Date </span>
                                <span> Description </span>
                            </li>
                        </ul>
                    </div>
                ) : ""
            }
            {
                myMapsPanel ? (
                    <div className="mapPanel">
                        <h2>My Maps</h2>
                        <ul>
                            <li className="map">
                                <strong> Map </strong>
                                <img src={battle} width="200"/>
                                <span> Date </span>
                                <span> Description </span>
                            </li>

                            <li className="map">
                                <strong> Map </strong>
                                <img src={battle} width="200"/>
                                <span> Date </span>
                                <span> Description </span>
                            </li>
                        </ul>
                    </div>
                ) : ""
            }

            
        </>
    )
}
