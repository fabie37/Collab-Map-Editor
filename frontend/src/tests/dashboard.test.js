import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from '../pages/Dashboard';
//import Link​​ from 'react-router-dom/Link';
let BrowserRouter = require('react-router-dom').BrowserRouter;
let Link = require('react-router-dom').Link;
let Route = require('react-router-dom').Route;
require('regenerator-runtime/runtime');

test('Dashboard renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Route path="/" exact component={Dashboard} />
      </BrowserRouter>
    )
    .toJSON();
  // put the existing a link here instead of creating a sample facebook one
  expect(tree).toMatchSnapshot();
});
