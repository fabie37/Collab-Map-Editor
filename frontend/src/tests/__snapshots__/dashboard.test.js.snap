// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <div
      className="filter-panel"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          aria-label={null}
          className="dropdown-toggle btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Filter
        </button>
        <div
          aria-hidden={true}
          className="dropdown-menu"
          role="menu"
          tabIndex="-1"
        >
          <button
            className="dropdown-item active"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            All Sports
          </button>
          <button
            className="dropdown-item"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            My Events
          </button>
          <button
            className="dropdown-item"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            Running
          </button>
          <button
            className="dropdown-item"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            Cycling
          </button>
          <button
            className="dropdown-item"
            color="primary"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            Swimming
          </button>
        </div>
      </div>
    </div>
    <ul
      className="events-list"
    />
    
    
  </div>
</div>
`;
